source/common/router/config_impl.cc

```diff
@@ -486,15 +487,15 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
       cluster_not_found_response_code_(ConfigUtility::parseClusterNotFoundResponseCode(
           route.route().cluster_not_found_response_code())),
       timeout_(PROTOBUF_GET_MS_OR_DEFAULT(route.route(), timeout, DEFAULT_ROUTE_TIMEOUT_MS)),
-      idle_timeout_(PROTOBUF_GET_OPTIONAL_MS(route.route(), idle_timeout)),
+      idle_timeout_(PROTOBUF_GET_MS_OR_DEFAULT(route.route(), idle_timeout, -1)),
```

Comment:
can you use a constexpr rather than -1 here?

Reply:
Done